src/app/rpmostree-builtin-kargs.cxx

```diff
@@ -355,7 +358,17 @@ rpmostree_builtin_kargs (int            argc,
                                                 error))
     return FALSE;
 
-  g_print("Kernel arguments updated.\nRun \"systemctl reboot\" to start a reboot\n");
+  if (opt_unchanged_exit_77)
+    {
+      if (!rpmostree_has_new_default_deployment (os_proxy, previous_deployment))
+        {
+          invocation->exit_code = RPM_OSTREE_EXIT_UNCHANGED;
+          return TRUE;
+        }
+    }
+
+  if (rpmostree_has_new_default_deployment (os_proxy, previous_deployment))
+    g_print("Kernel arguments updated.\nRun \"systemctl reboot\" to start a reboot\n");
```

Comment:
This is good as is, but another more succinct way to write this is:

```
if (rpmostree_has_new_default_deployment (...))
  g_print (...)
else if (opt_unchanged_exit_77)
  invocation->exit_code = RPM_OSTREE_...
```

Reply:
I think this would still be nice to do in a follow-up so that we don't have to call `rpmostree_has_new_default_deployment` twice in the case the kargs changed.